TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/WorkspacesController.php

```diff
@@ -491,6 +506,144 @@ protected function computeSiteChanges(Workspace $selectedWorkspace)
     }
 
     /**
+     * Retrieves the given node's corresponding node in the base workspace (that is, which would be overwritten if the
+     * given node would be published)
+     *
+     * @param NodeInterface $modifiedNode
+     * @return NodeInterface
+     */
+    protected function getOriginalNode(NodeInterface $modifiedNode)
+    {
+        $baseWorkspaceName = $modifiedNode->getWorkspace()->getBaseWorkspace()->getName();
+        if (!isset($this->contextContexts[$baseWorkspaceName])) {
+            $contextProperties = $modifiedNode->getContext()->getProperties();
+            $contextProperties['workspaceName'] = $baseWorkspaceName;
+            $this->contextContexts[$baseWorkspaceName] = $this->contextFactory->create($contextProperties);
```

Comment:
If my workspace contains changes to nodes having differing dimension values, the first node will populate the cache and the second would have it's original node fetched from the "wrong" workspace… no?